daily-challenges/2024-09/14/hautvph17241.spec.ts

```diff
@@ -0,0 +1,32 @@
+function calculateBMI(height: number, weight: number) {
+  const BMI = weight / (height * height);
+  let classify: any;
+  if (BMI < 18.5) {
+    classify = "Gầy";
+  } else if (18.5 <= BMI && BMI < 24.9) {
+    classify = "Bình thường";
+  } else if (25 <= BMI && BMI < 29.9) {
+    classify = "Thừa cân";
+  } else {
+    classify = "Béo phì";
+  }
+  return [BMI, classify];
+}
+const [BMI, classify] = calculateBMI(1.68, 63);
+console.log("Kết quả BMI: " + BMI);
+console.log("Phân loại: " + classify);
+
+import { test, expect } from "@playwright/test";
+const userName = "Hautvph17241";
+const email = "hautvph17241@gmail.com";
+test("Day 14/08/2024", async ({ page }) => {
+  await page.goto("https://material.playwrightvn.com/");
+  await page.waitForLoadState("networkidle");
+  await page.getByRole("link", { name: "Register Page" }).click();
+  await page.locator("#username").fill(userName);
+  await page.locator("#email").fill(email);
+  await page.getByRole("button", { name: "Register" }).click();
+  const tableElements = page.locator('//table[@id="userTable"]/tbody/tr/td');
+  expect(await tableElements.nth(1).innerText()).toEqual(userName);
```

Comment:
Not required, but should do:
Đối với các mệnh đề await, theo mình không nên cho vào trong expect, if, for hay các block lồng tương tự.
Khi có lỗi xảy ra, rất khó để cắm debug.

```
const tableText = await tableElements.....
expect(tableText).toEqual(smth)
```

Reply:
em đã sửa rồi anh xem lại giúp em, em cảm ơn a nhiều !!!